presto-main/src/main/java/com/facebook/presto/metadata/DiscoveryNodeManager.java

```diff
@@ -230,7 +230,7 @@ private synchronized void refreshNodesInternal()
         // TODO: make it a whitelist (a failure-detecting service selector) and maybe build in support for injecting this in airlift
         Set<ServiceDescriptor> services = serviceSelector.selectAllServices().stream()
                 .filter(service -> !failureDetector.getFailed().contains(service))
-                .filter(service -> nodeStatusService.isAllowed(service.getNodeId()))
+                .filter(service -> !nodeStatusService.isPresent() || nodeStatusService.get().isAllowed(service.getNodeId()) || currentNode.getNodeIdentifier().compareTo(service.getNodeId()) == 0)
```

Comment:
can you clarify why do we need `currentNode.getNodeIdentifier().compareTo(...)`?

Reply:
Given coordinator won't be part of the worker tier, putting this comparison will count coordinator host as well here.

Reply:
Can you add a comment for this - it is not obvious by the look of it